### PR TITLE
Add `scarb doc`-generated docs to GitHub Pages

### PR DESCRIPTION
**Pull request by DelevoXDG**
Closes https://github.com/software-mansion/scarb/issues/1596

## Introduced changes

- Update docs deployment workflow to add subpages for `snforge_std` and `sncast_std` docs generated by `scarb doc`
  - Include said docs under `/snforge_std` and `/sncast_std`
  - You can preview at https://delevoxdg.github.io/starknet-foundry/appendix/snforge-library.html
- Update docs and READMEs to link these

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/book.toml [hunk 1]
```diff
@@ -10,6 +10,10 @@ description = "Starknet Foundry is a toolchain for developing Starknet smart con
 warning-policy = "error"
 follow-web-links = true
 traverse-parent-directories = false
+exclude = [
+    "https://foundry-rs.github.io/starknet-foundry/snforge_std/",
+    "https://foundry-rs.github.io/starknet-foundry/sncast_std/"
+]
```
Comment (DelevoXDG) [inline docs/book.toml:16]: This is only necessary for the first deployment, later this can be removed

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -35,19 +35,42 @@ jobs:
       - uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a
         with:
           toolchain: stable
+      - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84
```
Comment (DelevoXDG) [inline .github/workflows/docs.yml:38]: Although this is unrelated, I'm just not sure why this particular workflow doesn't have cache 🤔

**Review by maciektr — APPROVED** [review]
Great work! Can't wait to see it in use :D 

Leaving for the Foundry team to review, nothing really to add from me :) 

Can you please make sure it supports the highlighting though? 🤔 
https://docs.swmansion.com/scarb/docs/extensions/documentation-generation.html#cairo-code-highlighting-using-mdbook

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -35,19 +35,42 @@ jobs:
...
       - uses: actions/checkout@v4
       - name: Install mdBook
         run: |
           cargo install --version ${MDBOOK_VERSION} mdbook
       - name: Install mdBook Link-Check
         run: |
           cargo install mdbook-linkcheck
+      - name: Install Scarb
+        uses: software-mansion/setup-scarb@v1.3.3
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply .github/workflows/docs.yml]: Sure, just did that based on the way it's done in other workflows here, I suppose we should change it elsewhere as well for consistency?

[the remark this answers is not in the document]
Comment (maciektr) [reply .github/workflows/docs.yml]: I am not the owner of this repo, but I would advise that :)

**Review by maciektr — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 2 above)
Comment (cptartur) [reply .github/workflows/docs.yml:38]: I think we might have just forgotten to add one

**Review by cptartur — COMMENTED** [review]
This is a good initial implementation. I think some more effort on our side will be required to have it replace our current docs of `snforge_std` and `sncast_std`. 

I wonder though if we should already include links to the docs generated by this in `docs/src/appendix/sncast-library.md` and `docs/src/appendix/snforge-library.md` or should we wait to do some additional cleanup of the docs.

cc @kkawula @franciszekjob

**Comment by maciektr** [comment]
> I wonder though if we should already include links to the docs generated by this in docs/src/appendix/sncast-library.md and docs/src/appendix/snforge-library.md or should we wait to do some additional cleanup of the docs.

Why not? Making it discoverable through auto-generated docs is the whole point.

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -35,19 +35,47 @@ jobs:
...
+      - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84
       - uses: actions/checkout@v4
       - name: Install mdBook
         run: |
           cargo install --version ${MDBOOK_VERSION} mdbook
       - name: Install mdBook Link-Check
         run: |
           cargo install mdbook-linkcheck
+      - name: Install Scarb
+        uses: software-mansion/setup-scarb@v1
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply .github/workflows/docs.yml:47]: https://github.com/DelevoXDG/scarb/actions/runs/11443488714/job/31836249298

Tried doing that for corelib docs generation already. For now it fails, reason tbd, I'll check if it's the same with foundry.

[the remark this answers is not in the document]
Comment (DelevoXDG) [reply .github/workflows/docs.yml:47]: There was no issue https://github.com/DelevoXDG/starknet-foundry/actions/runs/11443688802/workflow, so we can use nightlies for generating docs for these libs. Although I wonder if it's possible at any point that scarb doc included with nightly versions can be incompatible with previous stable versions of Scarb? @wawel37

[the remark this answers is not in the document]
Comment (maciektr) [reply .github/workflows/docs.yml:47]: @DelevoXDG We can change this workflow to rely on stable after 2.9 is out. Right now, it's worth running the nightly. In general, nightly should not introduce breaking changes (until Scarb does with the next release of course), but may include some bugs.

[the remark this answers is not in the document]
Comment (wawel37) [reply .github/workflows/docs.yml:47]: Ideally, we want to keep the scarb doc as much backward compatible as we can. Tbh I don't have any example of a future change, that could make such a breaking change. As @maciektr  said, i think you can use the nightly for now. In worst case scenario, it could contain some bugs

[the remark this answers is not in the document]
Comment (cptartur) [reply .github/workflows/docs.yml:47]: Please add an issue to replace this with a stable version once it's out.

**Review by cptartur — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (maciektr) [reply .github/workflows/docs.yml:47]: Done: https://github.com/foundry-rs/starknet-foundry/issues/2607
